src/test/java/org/apache/sling/distribution/agent/impl/PrivilegeDistributionRequestAuthorizationStrategyTest.java

```diff
@@ -97,7 +97,34 @@ public void testPermissionOnAdd() throws Exception {
         when(resourceResolver.adaptTo(Session.class)).thenReturn(session);
         String[] paths = new String[]{"/foo"};
         for (String path : paths) {
-            when(acm.hasPrivileges(path, new Privilege[]{privilege, jcrReadPrivilege})).thenReturn(true);
+            when(acm.hasPrivileges(path, new Privilege[]{jcrReadPrivilege, privilege})).thenReturn(true);
+        }
+        when(distributionRequest.getPaths()).thenReturn(paths);
+
+        when(distributionRequest.getRequestType()).thenReturn(DistributionRequestType.ADD);
+        strategy.checkPermission(resourceResolver, distributionRequest);
+    }
+    
+    @Test
+    public void testAdditionalPermissionsOnAdd() throws Exception {
+        additionalJcrPrivilegesForAdd = new String[] {"addPermission"};
+        PrivilegeDistributionRequestAuthorizationStrategy strategy = new PrivilegeDistributionRequestAuthorizationStrategy(jcrPrivilege, additionalJcrPrivilegesForAdd, additionalJcrPrivilegesForDelete);
+        DistributionRequest distributionRequest = mock(DistributionRequest.class);
+        ResourceResolver resourceResolver = mock(ResourceResolver.class);
+        Session session = mock(Session.class);
+        AccessControlManager acm = mock(AccessControlManager.class);
+        Privilege privilege = mock(Privilege.class);
+        when(acm.privilegeFromName(jcrPrivilege)).thenReturn(privilege);
+        Privilege additionalPrivilegeForAdd = mock(Privilege.class);
+        for (String privilegeAdd : additionalJcrPrivilegesForAdd) {
```

Comment:
either this `for` loop should be dropped, or the test adjusted to actually create correct fixtures. Right now, if one were to add another permission to `additionalJcrPrivilegesForAdd` array, the fixture below will return _identical_ mock for both of them. (not to mention the test will likely error-out because the fixture for `acm.hasPrivileges` won't work due to unexpected arguments).
Similarly, if you decide that `for` loop _IS NOT_ to be dropped, use `Answer` to verify the arguments sent while setting up the fixture for `acm.hasPrivileges`